code/modules/mob/dead/new_player/latejoin_menu.dm

```diff
@@ -0,0 +1,192 @@
+				"prioritized" = (job_datum in SSjob.prioritized_jobs),
+				"used_slots" = job_datum.current_positions,
+				"open_slots" = job_datum.total_positions < 0 ? "∞" : job_datum.total_positions,
+			)
+
+			if(job_availability != JOB_AVAILABLE)
+				job_data["unavailable_reason"] = get_job_unavailable_error_message(job_availability, job_datum.title)
+
+			if(job_datum.total_positions < 0)
+				department_data["open_slots"] = "∞"
+
+			if(department_data["open_slots"] != "∞")
+				department_data["open_slots"] += job_datum.total_positions - job_datum.current_positions
+
+			department_jobs[job_datum.title] = job_data
+
+	return data
+
+/datum/latejoin_menu/ui_static_data(mob/user)
+	var/list/departments = list()
+
+	for(var/datum/job_department/department as anything in SSjob.joinable_departments)
+		var/list/department_jobs = list()
+		var/list/department_data = list(
+			"jobs" = department_jobs,
+			"color" = department.ui_color,
+		)
+		departments[department.department_name] = department_data
+
+		for(var/datum/job/job_datum as anything in department.department_jobs)
+			var/datum/outfit/outfit = job_datum.outfit
+			var/datum/id_trim/trim = initial(outfit.id_trim)
+
+			var/list/job_data = list(
+				"command" = !!(job_datum.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND),
+				"description" = job_datum.description,
+				"icon" = initial(trim.orbit_icon)
+			)
+
+			department_jobs[job_datum.title] = job_data
+
+	return list("departments_static" = departments)
+
+/datum/latejoin_menu/ui_status(mob/user)
+	return isnewplayer(user) ? UI_INTERACTIVE : UI_CLOSE
+
+/datum/latejoin_menu/ui_state(mob/user)
+	return GLOB.always_state
+
+/datum/latejoin_menu/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+
+	if(!ui.user.client || ui.user.client.interviewee || !isnewplayer(ui.user))
+		return TRUE
+
+	var/mob/dead/new_player/owner = ui.user
+
+	switch(action)
+		if("SelectedJob")
+			if(params["job"] == "Random")
+				var/list/dept_data = list()
+				for(var/datum/job_department/department as anything in SSjob.joinable_departments)
+					for(var/datum/job/job_datum as anything in department.department_jobs)
+						if(owner.IsJobUnavailable(job_datum.title, latejoin = TRUE) != JOB_AVAILABLE)
+							continue
+						dept_data += job_datum.title
+
+				if(dept_data.len <= 0) //Congratufuckinglations
+					tgui_alert(owner, "There are literally no random jobs available for you on this server, ahelp for assistance.", "Oh No!")
+					return TRUE
+
+				var/random_job
+
+				while(random_job != JOB_CHOICE_YES)
+					if(dept_data.len <= 0)
+						tgui_alert(owner, "It seems that there are no more random jobs available for you!", "Oh No!")
+						return TRUE
+
+					var/random = pick_n_take(dept_data)
+					var/list/random_job_options = list(JOB_CHOICE_YES, JOB_CHOICE_CANCEL)
+
+					if(dept_data.len > 0)
+						random_job_options.Insert(2, JOB_CHOICE_REROLL) // Add reroll in the middle, it makes more sense, I think.
```

Comment:
Why are you doing it like this?

`var/list/random_job_options = list(JOB_CHOICE_YES, JOB_CHOICE_REROLL, JOB_CHOICE_CANCEL)`